docs/troubleshooting.md

```diff
@@ -46,6 +48,41 @@ This is typically encountered when you attempt to install BKPR with a DNS zone (
 
 ![Azure SP Conflict](images/azure-sp-conflict.png)
 
+## Troubleshooting BKPR ingress
+
+### Let's Encrypt
+
+BKPR deploys [`cert-manager`](#components.md/cert-manager) Kubernetes add-on to automate the management and issuance of TLS certificates. `cert-manager` delegates on Let's Encrypt to retrieve valid X.509 certificates used to protect designated Kubernetes ingress resources with TLS encryption for HTTP traffic.
+
+A Kubernetes ingress resource is designated to be TLS-terminated at the NGINX controller when the following annotations are present:
+
+```
+Annotations:
+  kubernetes.io/ingress.class:                        nginx
+  kubernetes.io/tls-acme:                             true
+```
+
+`cert-manager` watches for Kubernetes ingress resources. When an ingress resource with the `"kubernetes.io/tls-acme": true` annotation is seen for the first time, `cert-manager` tries to issue an [ACME certificate using HTTP-01 challenge](http://docs.cert-manager.io/en/latest/tutorials/acme/http-validation.html). This works by creating a temporary ingress resource name like `cm-acme-http-solver-${pod_id}` and then triggering the HTTP-01 challenge protocol. This requires that Let's Encrypt is able to resolve the DNS name associated with the Kubernetes ingress resource. On a Kubernetes cluster where BKPR is deployed, this requires that External DNS is functioning properly.
+
+During the time while `cert-manager` negotiates the certificate issue with Let's Encrypt, NGNIX controller temporatily uses a self-signed certificate. This situation should be transient and when it lasts for more than a couple of minutes there is a problem that prevents the certificate from being issued. Please note that when using the staging environment for Let's Encrypt the certificates issued by it not be recognized as valid (the signing CA is not a recognized one). 
```

Comment:
temporatily -> typo. Could you please run a spell check across the doc?

Reply:
when it lasts for more than a couple of minutes there is a problem that prevents the certificate from being issued -> when it lasts for more than a few minutes, it implies that there is a problem preventing the certificate from being issued.

Reply:
there is a problem -> this is slightly vague, do we want to indicate what those problems could be?

Reply:
"Please note that when using the staging environment for Let's Encrypt the certificates issued by it not be recognized as valid (the signing CA is not a recognized one). " -> I would actually put this as a note at the top of this section, so that users who are using the staging environment understand immediately that their issue is probably because they are not running as per production requirements.